apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSinkTest.java

```diff
@@ -0,0 +1,253 @@
+        doThrow(DataTransformer.TransformationException.class)
+            .when(sinkSpy)
+            .transformMessage(apiVersion, message);
+      } else {
+        doReturn(createChangeClaimFromMessage(message))
+            .when(sinkSpy)
+            .transformMessage(apiVersion, message);
+      }
+    }
+
+    doNothing()
+        .when(sinkSpy)
+        .writeError(anyString(), anyString(), any(DataTransformer.TransformationException.class));
+
+    List<RdaChange<String>> expected =
+        messages.stream().map(this::createChangeClaimFromMessage).collect(Collectors.toList());
+
+    Wrapper<List<RdaChange<String>>> wrapper = new Wrapper<>();
+
+    assertThrows(
+        DataTransformer.TransformationException.class,
+        () -> wrapper.wrap(sinkSpy.transformMessages(apiVersion, messages)));
+
+    verify(sinkSpy, times(1))
+        .writeError(
+            eq(apiVersion), eq(badMessage), any(DataTransformer.TransformationException.class));
+  }
+
+  /**
+   * Helper method to create a {@link RdaChange} object with the given message.
+   *
+   * @param message The message to use to create the {@link RdaChange} object.
+   * @return The created {@link RdaChange} object.
+   */
+  private RdaChange<String> createChangeClaimFromMessage(String message) {
+    return new RdaChange<>(0, RdaChange.Type.UPDATE, message + "_claim", Instant.ofEpochMilli(1));
+  }
+
+  /**
+   * Helper method to check if the two {@link RdaChange} object lists are equivalent.
+   *
+   * @param expected The expected list of {@link RdaChange} values.
+   * @param actual The actual list of {@link RdaChange} values.
+   * @param <T> The type of the embedded data of the {@link RdaChange} objects.
+   */
+  private <T> void assertClaimChangesEquals(
+      List<RdaChange<T>> expected, List<RdaChange<T>> actual) {
+    assertEquals(expected.size(), actual.size());
+
+    for (int i = 0; i < expected.size(); ++i) {
+      assertClaimChangeEquals(expected.get(i), actual.get(i));
+    }
+  }
+
+  /**
+   * Helper method to check if the two {@link RdaChange} objects are equivalent.
+   *
+   * @param expected The expected {@link RdaChange} value.
+   * @param actual The actual {@link RdaChange} value.
+   * @param <T> The type of the embedded data of the {@link RdaChange} object.
+   */
+  private <T> void assertClaimChangeEquals(RdaChange<T> expected, RdaChange<T> actual) {
+    assertEquals(expected.getSequenceNumber(), actual.getSequenceNumber());
+    assertEquals(expected.getType(), actual.getType());
+    assertEquals(expected.getClaim(), actual.getClaim());
+    assertEquals(expected.getTimestamp(), actual.getTimestamp());
+  }
+
+  /**
+   * Helper class for retrieving values from inside lambda expressions.
+   *
+   * @param <T> The type of data being wrapped.
+   */
+  static class Wrapper<T> {
+
+    private T wrapped;
+
+    void wrap(T wrapped) {
+      this.wrapped = wrapped;
+    }
+
+    T unwrap() {
+      return wrapped;
+    }
+  }
```

Comment:
Wouldn't `AtomicReference` do the same thing without the need to define a new class?

Reply:
Ah, I didn't know this one, thanks!